test/reverseRegistrar/TestL2ReverseRegistrar.js

```diff
@@ -0,0 +1,289 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const { namehash } = require('../test-utils/ens')
+const { EMPTY_ADDRESS } = require('../test-utils/constants')
+
+describe('L2ReverseRegistrar', function () {
+  let L2ReverseRegistrar
+  let L2ReverseRegistrarWithAccount2
+  let MockSmartContractWallet
+  let MockOwnable
+  let signers
+  let account
+  let account2
+  let setNameForAddrWithSignatureFuncSig =
+    'setNameForAddrWithSignature(address,string,uint256,bytes)'
+  let setNameForAddrWithSignatureAndOwnableFuncSig =
+    'setNameForAddrWithSignatureAndOwnable(address,address,string,uint256,bytes)'
+  let setTextForAddrWithSignatureFuncSig =
+    'setTextForAddrWithSignature(address,string,string,uint256,bytes)'
+  let setTextForAddrWithSignatureAndOwnableFuncSig =
+    'setTextForAddrWithSignatureAndOwnable(address,address,string,string,uint256,bytes)'
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    account = await signers[0].getAddress()
+    account2 = await signers[1].getAddress()
+
+    const L2ReverseRegistrarFactory = await ethers.getContractFactory(
+      'L2ReverseRegistrar',
+    )
+    L2ReverseRegistrar = await L2ReverseRegistrarFactory.deploy(
+      namehash('optimism.reverse'),
+    )
+
+    const MockSmartContractWalletFactory = await ethers.getContractFactory(
+      'MockSmartContractWallet',
+    )
+    MockSmartContractWallet = await MockSmartContractWalletFactory.deploy(
+      account,
+    )
+
+    const MockOwnableFactory = await ethers.getContractFactory('MockOwnable')
+    MockOwnable = await MockOwnableFactory.deploy(
+      MockSmartContractWallet.address,
+    )
+
+    L2ReverseRegistrarWithAccount2 = L2ReverseRegistrar.connect(signers[1])
+
+    await L2ReverseRegistrar.deployed()
+  })
+
+  beforeEach(async () => {
+    result = await ethers.provider.send('evm_snapshot')
+  })
+  afterEach(async () => {
+    await ethers.provider.send('evm_revert', [result])
+  })
+
+  it('should deploy the contract', async function () {
+    expect(L2ReverseRegistrar.address).to.not.equal(0)
+  })
+
+  describe('setName', () => {
+    it('should set the name record for the calling account', async function () {
+      const name = 'myname.eth'
+      const tx = await L2ReverseRegistrar.setName(name)
+      await tx.wait()
+
+      const node = await L2ReverseRegistrar.node(
+        await ethers.provider.getSigner().getAddress(),
+      )
+      const actualName = await L2ReverseRegistrar.name(node)
+      expect(actualName).to.equal(name)
+    })
+  })
+
+  describe('setNameForAddrWithSignature', () => {
+    it('allows an account to sign a message to allow a relayer to claim the address', async () => {
+      const funcId = ethers.utils
+        .id(setNameForAddrWithSignatureFuncSig)
+        .substring(0, 10)
+
+      const block = await ethers.provider.getBlock('latest')
+      const signatureExpiry = block.timestamp + 3600
```

Comment:
Isn't this supposed to be `inceptionDate` (and you need a test case to check the failure scenario when `inceptionDate = block.timestamp - 1`)

Reply:
Or rather you should have both inceptionDate and signatureExpiry support